packages/hyperion-autologging/src/ALSurface.ts

```diff
@@ -300,15 +294,15 @@ export function init(options: InitOptions): ALSurfaceHOC {
        */
       callFlowlet = new flowletManager.flowletCtor(surface, surfaceCtx.callFlowlet ?? flowletManager.root);
       callFlowlet.data.surface = nonInteractiveSurfacePath;
-      surfaceData = {
-        surface: surfacePath,
-        nonInteractiveSurface: nonInteractiveSurfacePath,
+      surfaceData = new ALSurfaceData(
+        surfacePath,
+        surfaceCtx,
+        nonInteractiveSurfacePath,
         callFlowlet,
+        capability,
         domAttributeName,
         domAttributeValue,
-        capability,
-      };
-      SurfaceDataMap.set(nonInteractiveSurfacePath, surfaceData);
```

Comment:
This is done in the constructor now?

Reply:
Yes, I wanted to make sure there is only want to manipulate the `Map` and also have some assertion mechanism to check the health of the `Map`. It was tricky to get all cases right.